examples/02_decoding/plot_oasis_vbm.py

```diff
@@ -39,132 +45,139 @@
-# reverse variance threshold
-coef = variance_threshold.inverse_transform(coef)
-# reverse masking
-weight_img = nifti_masker.inverse_transform(coef)
+######################################################################
+weight_img = decoder.coef_img_['beta']
 
 # Create the figure
 from nilearn.plotting import plot_stat_map, show
-bg_filename = gray_matter_map_filenames[0]
-z_slice = 0
-
+bg_filename = gm_imgs[0]
 
-fig = plt.figure(figsize=(5.5, 7.5), facecolor='k')
-# Hard setting vmax to highlight weights more
 display = plot_stat_map(weight_img, bg_img=bg_filename,
-                        display_mode='z', cut_coords=[z_slice],
-                        figure=fig, vmax=1)
-display.title('SVM weights', y=1.2)
-
-# Measure accuracy with cross validation
-from sklearn.model_selection import cross_val_score
-cv_scores = cross_val_score(anova_svr, gm_maps_masked, age)
-
-# Return the corresponding mean prediction accuracy
-prediction_accuracy = np.mean(cv_scores)
-print("=== ANOVA ===")
-print("Prediction accuracy: %f" % prediction_accuracy)
-print("")
+                        display_mode='z', cut_coords=[-6],
+                        title="Decoder r2: %g" % prediction_score)
+
+show()
 
-### Inference with massively univariate model #################################
+######################################################################
+# Visualize the quality of predictions
+import matplotlib.pyplot as plt
+plt.figure(figsize=(6, 4.5))
+plt.suptitle("Decoder: Mean Absolute Error %.2f years" % prediction_score)
+linewidth = 3
+plt.plot(age_test, label="True age", linewidth=linewidth)
+plt.plot(age_pred, '--', c="g", label="Predicted age", linewidth=linewidth)
+plt.ylabel("age")
+plt.xlabel("subject")
+plt.legend(loc="best")
+######################################################################
+plt.figure(figsize=(6, 4.5))
+plt.plot(age_test - age_pred, label="True age - predicted age",
+         linewidth=linewidth)
+plt.xlabel("subject")
+plt.legend(loc="best")
+###############################################################################
+# Comparing to massi-univariate analysis
+###############################################################################
 print("Massively univariate model")
 
+# First, we have to use the mask that we obtained after the variance
+# thresholding.
+nifti_masker = NiftiMasker(mask_img=mask)
+gm_imgs_train_masked = nifti_masker.fit_transform(gm_imgs_train)
+
 # Statistical inference
 from nilearn.mass_univariate import permuted_ols
-data = variance_threshold.fit_transform(gm_maps_masked)
 neg_log_pvals, t_scores_original_data, _ = permuted_ols(
-    age, data,  # + intercept as a covariate by default
+    age_train, gm_imgs_train_masked,  # + intercept as a covariate by default
     n_perm=2000,  # 1,000 in the interest of time; 10000 would be better
     n_jobs=1)  # can be changed to use more CPUs
 signed_neg_log_pvals = neg_log_pvals * np.sign(t_scores_original_data)
```

Comment:
Maybe this paragraph need comment or mention of another example that explain the statistical inference in details ?